File: backend/src/main/java/cn/etarch/mao/session/entity/PermissionLevel.java
package cn.etarch.mao.session.entity;

public enum PermissionLevel {
    READ_ONLY,
    READ_WRITE,
    SMART,
    FULL;

    public static PermissionLevel fromString(String value) {
        if (value == null) return READ_ONLY;
        try {
            return valueOf(value);
        } catch (IllegalArgumentException e) {
            return READ_ONLY;
        }
    }
}

File: backend-ts/src/harness/tool/permission-level.ts
export type PermissionLevel = 'READ_ONLY' | 'READ_WRITE' | 'SMART' | 'FULL';

export function permissionLevelFromString(value: string | null | undefined): PermissionLevel {
  if (value == null) return 'READ_ONLY';
  if (value === 'READ_ONLY' || value === 'READ_WRITE' || value === 'SMART' || value === 'FULL') {
    return value;
  }
  return 'READ_ONLY';
}
